### PR TITLE
tools/simulator: support deleting a specified store

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Ref #8135

### What is changed and how does it work?

support deleting a specified store

can execute `curl --location --request POST 'http://127.0.0.1:20180/event?event=down-node&node-id=6'`

for example, in `balance-leader` case

image
image

```commit-message
```

### Check List

Tests

- Integration test
- Manual test (add detailed scripts or steps below)

### Release note

```release-note
None.
```

### CODE REVIEW

File: tools/pd-simulator/simulator/cases/balance_leader.go [hunk 1]
```diff
@@ -58,10 +61,18 @@ func newBalanceLeader(config *sc.SimConfig) *Case {
 		})
 	}
 
-	simCase.Checker = func(regions *core.RegionsInfo, _ []info.StoreStats) bool {
-		for i := 1; i <= totalStore; i++ {
-			leaderCount := regions.GetStoreLeaderCount(uint64(i))
-			if !isUniform(leaderCount, totalRegion/totalStore) {
+	simCase.Checker = func(stores []*metapb.Store, regions *core.RegionsInfo, _ []info.StoreStats) bool {
```
Comment (rleungx) [inline tools/pd-simulator/simulator/cases/balance_leader.go:64]: Do we still needs this checker?

**Review by rleungx — APPROVED** [review]
LGTM except for the lint issue.

File: tools/pd-simulator/simulator/cases/balance_leader.go (hunk 1 above)
Comment (HuSharp) [reply tools/pd-simulator/simulator/cases/balance_leader.go:64]: yes, I think we can keep the logic for previous cases

**Review by okJiang — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Review by niubell — APPROVED** [review]

**Pull request merged**